### PR TITLE
RM#48185: changed irrecovable move generation in order to be created fr…

### PR DESCRIPTION
**Pull request by jdo-axelor**
…om invoice move and not invoice tax lines, it correct holdback and currency problems

### CODE REVIEW

File: axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java [hunk 1]
```diff
@@ -872,102 +875,87 @@ public Move createIrrecoverableMove(
...
+        Reconcile reconcile =
+            reconcileService.createReconcile(invoiceMoveLine, creditMoveLine, amount, false);
+        if (reconcile != null) {
+          reconcileService.confirmReconcile(reconcile, true);
+        }
+      } else {
+        amount =
+            invoiceMoveLine.getCredit().multiply(prorataRate).setScale(2, RoundingMode.HALF_UP);
+        if (invoiceMoveLine.getAccount().getAccountType().getTechnicalTypeSelect()
+            == AccountTypeRepository.TYPE_TAX) {
```
Comment (jmi-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java:914]:
```suggestion
        if (AccountTypeRepository.TYPE_TAX
				.equals(invoiceMoveLine.getAccount().getAccountType().getTechnicalTypeSelect())) {
```

File: axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java [hunk 2]
```diff
@@ -872,102 +875,87 @@ public Move createIrrecoverableMove(
...
+
     if (isInvoiceReject) {
       creditAmount = invoice.getRejectMoveLine().getAmountRemaining();
-      debitAmount = creditAmount;
     } else {
       creditAmount = invoice.getCompanyInTaxTotalRemaining();
-      debitAmount = creditAmount;
     }
 
     // Debits MoveLines Tva
```
Comment (jmi-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java:885]: Deprecated comment

File: axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java [hunk 3]
```diff
@@ -872,102 +875,87 @@ public Move createIrrecoverableMove(
...
+                seq,
+                originStr,
+                invoice.getInvoiceId());
+        move.getMoveLineList().add(creditMoveLine);
+
+        Reconcile reconcile =
+            reconcileService.createReconcile(invoiceMoveLine, creditMoveLine, amount, false);
+        if (reconcile != null) {
+          reconcileService.confirmReconcile(reconcile, true);
+        }
```
Comment (jmi-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java:909]: It should not be done here, the InvoiceTerms have not been created on the creditMoveLine yet. 
Instead, it should be done after line 542, when we call moveValidateService.accounting(move);

**Review by jmi-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java (hunk 1 above)
Comment (jdo-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java]: thanks you i missed that it's a string and not a int

**Pull request merged**